Lenvill's Ignition Patch

Lenvill-Jarlsen opened the pull request:
Build.Gradle:
- Added a "sourceSets.all" block to make bugtesting easier
- Removed duplicate implementation of JEI

IIgnitable:
- Added IIgnitable interface for all blocks that can be ignited. Has a static "playIgnitionSound" method that plays a slightly randomized ignition sound at target block, and requires all classes it is applied to to implement a "fireItUp" function that other things can now reference to ignite those blocks.

For each of BlockBloomery, BlockFirepit, and BlockForge:
- Gave it the IIgnitable interface
- Annotated elements "onBlockActivated" method
- Reworked the section of "onBlockActivated" that handles ignition. Is now near identical across all blocks
- Added implementation of FireItUp, which checks all block-and-tile-related conditions influencing whether or not the IIgnitable can ignite, and if it can ignite, puts it into an ignited state.
- All blocks now play a pitch-shifted fire charge sound when ignited, instead of just the firepit
- Blocks only fire fireItUp on the server side to avoid the false start bug when the random success chance rolled a success on the client side but failed on the server side
- If clicked on with an instance of IIgniter that has its "canLight" boolean set to false, it will end the interact process. I did this intentionally to make implimentinng support for pyrotech's weird lighters easier, but it can also be put on any item if it should not open up the GUI of the block in question

BlockBloomery:
- Now has ignition sound
- If unable to be ignited, right clicking on it with an igniter will open the GUI
- Fixed ignition check to solve the issues of the bloomery being ignited and stuck in an active state if ignited when totally empty, and entering a false start when fuel and ore didn't match.

BlockFirepit
- Changed unsuccessful ignition sound to the same flint and steel click as other blocks
- If clicked on by an igniter while lacking fuel, will precent the igniter from trying to set fire to an adjacent block
- If an attempt to ignite it in the rain is made, it will make a sizzle noise without ever entering an active state

BlockForge
- Can no longer be activated when its top is exposed to the rain. It will make a sizzle if the presence of rain prevents its activation
- No longer flashes when ignited when it has no fuel in it
- Auto-igniting forges now make a noise when being ignited and when being extinguished.

ItemLighter
- tryUse now returns a 0 if the ILighter canLlight boolean is false. Used to bypass normal ignition steps.
- Added @Nonnull annotation to EnumActionResult because IntelliJ told me to
- ILighters now only light fires on the server side. This is done to fix the bug where a client-side-only fire would briefly appear when it rolled a successful ignition on the client but a failed ignition on the server

TileEntityBloomery
- Renamed isActive() to getIsActive()
- Added setIsActive method
- added setProgressMax method
- Added check to Update() method to set isActive to false if it was true but it has no fuel.
- Removed AP's old light() function, as the ignition process is now handled on the block side.

TileEntityFirepit
- Renamed isLit() to getIsLit()
- Renamed isLit() to setIsLit()
- changed isWet from private to public
- tryLight() uses fireItUp() to handle igition

TileEntityForge
- Added getFuel() method
- Removed the fireUpForge() method, as that is now handled by the fireItUp() method in the block class
- Changed addFuel to use fireItUp() when a hot fuel is added to the forge, which means it will play the whoosh noise

Lenvill-Jarlsen committed:
Lenvill's Ignition Patch

Build.Gradle:
- Added a "sourceSets.all" block to make bugtesting easier
- Removed duplicate implementation of JEI

IIgnitable:
- Added IIgnitable interface for all blocks that can be ignited. Has a static "playIgnitionSound" method that plays a slightly randomized ignition sound at target block, and requires all classes it is applied to to implement a "fireItUp" function that other things can now reference to ignite those blocks.

For each of BlockBloomery, BlockFirepit, and BlockForge:
- Gave it the IIgnitable interface
- Annotated elements "onBlockActivated" method
- Reworked the section of "onBlockActivated" that handles ignition. Is now near identical across all blocks
- Added implementation of FireItUp, which checks all block-and-tile-related conditions influencing whether or not the IIgnitable can ignite, and if it can ignite, puts it into an ignited state.
- All blocks now play a pitch-shifted fire charge sound when ignited, instead of just the firepit
- Blocks only fire fireItUp on the server side to avoid the false start bug when the random success chance rolled a success on the client side but failed on the server side
- If clicked on with an instance of IIgniter that has its "canLight" boolean set to false, it will end the interact process. I did this intentionally to make implimentinng support for pyrotech's weird lighters easier, but it can also be put on any item if it should not open up the GUI of the block in question

BlockBloomery:
- Now has ignition sound
- If unable to be ignited, right clicking on it with an igniter will open the GUI
- Fixed ignition check to solve the issues of the bloomery being ignited and stuck in an active state if ignited when totally empty, and entering a false start when fuel and ore didn't match.

BlockFirepit
- Changed unsuccessful ignition sound to the same flint and steel click as other blocks
- If clicked on by an igniter while lacking fuel, will precent the igniter from trying to set fire to an adjacent block
- If an attempt to ignite it in the rain is made, it will make a sizzle noise without ever entering an active state

BlockForge
- Can no longer be activated when its top is exposed to the rain. It will make a sizzle if the presence of rain prevents its activation
- No longer flashes when ignited when it has no fuel in it
- Auto-igniting forges now make a noise when being ignited and when being extinguished.

ItemLighter
- tryUse now returns a 0 if the ILighter canLlight boolean is false. Used to bypass normal ignition steps.
- Added @Nonnull annotation to EnumActionResult because IntelliJ told me to
- ILighters now only light fires on the server side. This is done to fix the bug where a client-side-only fire would briefly appear when it rolled a successful ignition on the client but a failed ignition on the server

TileEntityBloomery
- Renamed isActive() to getIsActive()
- Added setIsActive method
- added setProgressMax method
- Added check to Update() method to set isActive to false if it was true but it has no fuel.
- Removed AP's old light() function, as the ignition process is now handled on the block side.

TileEntityFirepit
- Renamed isLit() to getIsLit()
- Renamed isLit() to setIsLit()
- changed isWet from private to public
- tryLight() uses fireItUp() to handle igition

TileEntityForge
- Added getFuel() method
- Removed the fireUpForge() method, as that is now handled by the fireItUp() method in the block class
- Changed addFuel to use fireItUp() when a hot fuel is added to the forge, which means it will play the whoosh noise

ThatPolishKid99 commented on src/main/java/minefantasy/mfr/api/crafting/IIgnitable.java:
```diff
@@ -0,0 +1,26 @@
+     * @param pos BlockPos
+     * @param state IBlockState
+     */
+    void fireItUp(World world, BlockPos pos, IBlockState state);
```
I think changing this method name to 
igniteBlock() would be a good idea

Lenvill-Jarlsen replied:
That makes sense, will implement

ThatPolishKid99 commented on src/main/java/minefantasy/mfr/block/BlockBloomery.java:
```diff
@@ -112,25 +115,27 @@ public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, En
+			if (held.getItem() instanceof ItemFlintAndSteel || held.getItem() instanceof ILighter) {
+				if (!tile.getIsActive() && tile.getResult() != ItemStack.EMPTY) {
+					int uses = ItemLighter.tryUse(held, player);
+					if (uses != 0) // 1 for ignition, -1 for a failed attempt, 0 for a null input or for an item that needs to bypass normal ignition
```
I'd move this comment to above the ItemLighter.tryUse() method, so that the bracket of the if statement can be in the correct place and so its clear you're referring to the method's result

ThatPolishKid99 commented on src/main/java/minefantasy/mfr/block/BlockBloomery.java:
```diff
@@ -112,25 +115,27 @@ public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, En
-			// GUI
-			if (!world.isRemote && !tile.isActive() && !tile.hasBloom()) {
+			/// GUI
+			if (!world.isRemote && !tile.getIsActive() && !tile.hasBloom()) {
```
why rename this method?

Lenvill-Jarlsen replied:
I was originally planning on adding getters and setters for all the variables in each tile entity, but I scrapped that idea once I got a better idea of how everything was working and realized that getters and setters for everything was completely overkill. I changed the name of a couple of functions to be in line with this before deciding to scrap it - I can change it back if you would prefer it the old way

ThatPolishKid99 replied:
I think it would be good to change them back 

I just think isLit() is slightly easier to understand, because its essentially a question in the method, answered by the returned boolean

ThatPolishKid99 commented on src/main/java/minefantasy/mfr/block/BlockFirepit.java:
```diff
@@ -123,6 +125,7 @@ public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, En
 			boolean burning = firepit.isBurning();
 
 			if (!held.isEmpty()) {
+				/// Adding fuel
```
extremely minor nit here, why add an extra slash to the comment?

ThatPolishKid99 commented on src/main/java/minefantasy/mfr/block/BlockFirepit.java:
```diff
@@ -157,29 +161,18 @@ public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, En
+				} else {
+					if (held.getItem() instanceof ItemFlintAndSteel || held.getItem() instanceof ILighter) {
+						int uses = ItemLighter.tryUse(held, player);
+						if (uses != 0 && firepit.fuel > 0) // 1 for ignition, -1 for a failed attempt, 0 for a null input or for an item that needs to bypass normal ignition
```
Same as previous comment about this code comment

ThatPolishKid99 commented on src/main/java/minefantasy/mfr/block/BlockForge.java:
```diff
@@ -172,38 +172,48 @@ public boolean isBurning(IBlockAccess world, BlockPos pos) {
+				if (held.getItem() instanceof ItemFlintAndSteel || held.getItem() instanceof ILighter) {
+					if (!forge.getIsLit() && forge.getFuel() > 0 && forge.getTier() != 1) {
+						int uses = ItemLighter.tryUse(held, player);
+						if (uses != 0) // 1 for ignition, -1 for a failed attempt, 0 for a null input or for an item that needs to bypass normal ignition
```
Same as last two comments about this code comment :)

Lenvill-Jarlsen replied:
Oh - that's something I started doing in my own code to denote labels rather than just snippets that have been commented out. However, this isn't my codebase, so I understand why it feels out of place (especially given that one of my goals was to standardize things). I'll switch it to adhere to normal convention

ThatPolishKid99 reviewed (commented):
Generally very good, a couple of comments to address my concerns with more minor stuff

Lenvill-Jarlsen committed:
Lenvill's Ignition Patch (polished)

Renamed fireItUp to igniteBlock for added clarity.

Cleaned up my comments to better adhere to normal comment convention.

Lenvill-Jarlsen committed:
Reverted renaming

Reverted my changes to the names of certain functions
